File: tests/fixtures/projects/javascript/services.js
/**
 * Business logic.
 */
const { User, createUser } = require('./models');

class UserService {
    /**
     * Get a user.
     * @returns {User}
     */
    getUser() {
        return createUser("Alice");
    }
    
    /**
     * Process a user.
     * @param {User} user
     * @returns {string}
     */
    processUser(user) {
        return user.greet();
    }
}

module.exports = { UserService };

File: tests/fixtures/projects/rust/services.rs
//! Business logic.
use crate::models::{User, create_user};

pub struct UserService;

impl UserService {
    /// Get a user.
    pub fn get_user(&self) -> User {
        create_user("Alice")
    }
    
    /// Process a user.
    pub fn process_user(&self, user: &User) -> String {
        user.greet()
    }
}

